docs/contributors/code/release/README.md

```diff
@@ -0,0 +1,11 @@
+# Gutenberg Release Process
+
+The [Gutenberg repository](https://github.com/WordPress/gutenberg) on GitHub is used to perform several types of releases. This document serves as a checklist for each of these releases, and it can help you understand the different workflows involved.
```

Comment:
This README.md doesn't really serve as a checklist. I noticed it's coped from the plugin-release page, but I don't think it's relevant any more.

Or perhaps it should say "The linked documents serve as checklists".

TBH, the word 'checklist' is a bit confusing. 'Guide' is more appropriate IMO.